Jenkinsfile

```diff
@@ -0,0 +1,273 @@
+/* Set status on a github PR */
+def set_status(job, number, state, context){
+  lint_container.inside {
+    withCredentials([
+      string(
+        credentialsId: "rpc-jenkins-svc-github-pat",
+        variable: "GITHUB_PAT"
+      )
+    ]){
+      sh """#!/bin/bash
+        python2.7 rpc-gating/scripts/ghstatus.py ${job} ${number} ${state} ${context}
+      """
+    }
+  } // inside
+}
+
+/* This function returns a function that will be executed multiple times in
+* parallel. It builds a sub job, then links that sub-job to the pull request
+* that triggered this build.
+*/
+def makeBuildStep(job, parameters, context){
+  return {
+    def number = 0
+    def state = ""
+    def result = ""
+
+    print("Building ${job} with params: ${parameters}")
+    try{
+      result = build job: job, parameters: parameters, propagate: false
+    }catch (e){
+      print (e)
+    }
+   // To get a build number before the child job as finished, the build would
+   // have to be async, and I'm not sure how to get the final result in that
+   // case.
+   //set_status(job, insert_build_number_here, "pending", context)
+
+    print("Sub build ${job} result: ${result.result}")
+    if (result.result == "SUCCESS"){
+      state = "success"
+    } else {
+      state = "failure"
+      currentBuild.result = 'Failure'
+    }
+    // No point trying to update a PR if this was not triggered by a PR.
+    // CHANGE_ID is only set for PR builds, so will fail early for branch
+    // builds.
+    try {
+      print("Updating PR ${CHANGE_ID}")
+      try{
+        //can't get build number if the build aborted.
+        number = result.getNumber()
+        print("Sub build ${job} number: $number")
+      }catch (e) {
+        print("Failed to get build number for child job, Aborted?")
+      }
+      set_status(job, number, state, context)
+    }catch (e){
+      print("Failed to set PR status, maybe this job was not triggered by PR? ${e}")
+    }// try
+  } // closure
+} // func
+
+def determine_series_and_trigger(){
+  try {
+    series=CHANGE_TARGET
+    print("Using CHANGE_TARGET for series, value: ${env.CHANGE_TARGET}")
+    trigger="pr"
+    rpco_branch="origin/pr/${CHANGE_ID}/merge"
+    return [series, trigger, rpco_branch]
+  }catch (e){
+    rpco_branch=BRANCH_NAME
+    trigger="periodic"
+    print("Couldn't determine series using CHANGE_TARGET, trying containing branches.")
+
+    /* Determine series (mitaka, newton, master etc) by finiding the newest
+       branch that contains the first common commit. The problem is that some
+       branches/PRs contain multiple commits, so we start of looking at
+       HEAD^, then HEAD^^, HEAD^^^ etc until a branch match is found.
+    */
+
+    // Determine the series by finding the latest branch
+    // that contains the parent commit
+    containing_branches = sh(
```

Comment:
This block should be removed, as it was duplicated to L94. Will wait for other review comments before pushing another revision.